### ISSUE TITLE
ServerSelectionTimeoutError after start of mongo-connector

### ISSUE THREAD

**Issue by XDestination** [mongodb-labs/mongo-connector#653]
Below you can see the log output after starting mongo-connector.

> 2017-02-14 12:10:15,602 [ALWAYS] __main__:52 - Starting mongo-connector version: 2.5.0
2017-02-14 12:10:15,602 [ALWAYS] __main__:52 - Python version: 2.7.9 (default, Jun 29 2016, 13:08:31) 
[GCC 4.9.2]
2017-02-14 12:10:15,603 [ALWAYS] __main__:52 - Platform: Linux-3.16.0-4-amd64-x86_64-with-debian-8.7
2017-02-14 12:10:15,603 [ALWAYS] __main__:52 - pymongo version: 3.4.0
2017-02-14 12:10:15,618 [ALWAYS] __main__:52 - Source MongoDB version: 3.2.12
2017-02-14 12:10:15,618 [ALWAYS] __main__:52 - Target DocManager: mongo_connector.doc_managers.mongo_doc_manager version: 2.5.0
2017-02-14 12:10:45,835 [ERROR] mongo_connector.util:96 - Fatal Exception
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/mongo_connector/util.py", line 94, in wrapped
    func(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/mongo_connector/connector.py", line 398, in run
    hosts, replicaSet=repl_set)
  File "/usr/local/lib/python2.7/dist-packages/mongo_connector/connector.py", line 299, in create_authed_client
    client['admin'].authenticate(self.auth_username, self.auth_key)
  File "/usr/local/lib/python2.7/dist-packages/pymongo/database.py", line 1048, in authenticate
    connect=True)
  File "/usr/local/lib/python2.7/dist-packages/pymongo/mongo_client.py", line 500, in _cache_credentials
    writable_preferred_server_selector)
  File "/usr/local/lib/python2.7/dist-packages/pymongo/topology.py", line 214, in select_server
    address))
  File "/usr/local/lib/python2.7/dist-packages/pymongo/topology.py", line 189, in select_servers
    self._error_message(selector))
ServerSelectionTimeoutError: connection closed,connection closed,connection closed,connection closed

We are using a setup, where both the source and the target for the mongo-connector are two mongos running on the host (port 27017 and port 27817).

Any idea on how to fix this?

**Comment by ShaneHarvey** [comment]
It looks like PyMongo is unable to connect to the source cluster before attempting to authenticate. Can you post your config file and how you're launching mongo-connector? Be sure to remove any sensitive information.

**Comment by XDestination** [comment]
Sorry for the late response. The issue was caused by the implementation of apply_ssl in teh 2.5.0 release: https://github.com/mongodb-labs/mongo-connector/blob/2.5.0/mongo_connector/connector.py#L1063

The value of ssl.sslCertificatePolicy was not correctly applied and therefore the connection to one of our clusters failed. I fixed it by replacing the connector.py with the current version from the master branch.
